seed/shapesMigration/updateShapes.js

```diff
@@ -302,10 +446,55 @@ var unpublishApplication = function (app) {
   });
 };
 
-//
-// MAIN
-//
+/**
+ * Renews the jwt_login token if necessary.
+ *
+ * @returns {Promise}
+ */
+var renewJWTLogin = function () {
+  return new Promise(function(resolve, reject) {
+    var duration = moment.duration(new moment().diff(jwt_login_time)).asSeconds();
+    // less than 180 seconds left before token expiry
+    if (duration > jwt_expiry - 180) {
+      console.log('Requesting new login token');
+      return login(username, password)
+      .then(function () {
+        resolve()
+      })
+    } else {
+      resolve();
+    }
+  });
+}
+
+/**
+ * Returns an array of objects containing the optional batch parameters used by #getAllApplications.
+ * @param {number} applicationsCount
+ * @returns {array} batches array of objects used to facilitate calling #getAllApplications in batches.
+ */
+var getBatches = function (applicationsCount) {
```

Comment:
This is the only piece that feels a bit lackluster.  Im mainly using this batches array as a convenient way to run .reduce() later on and chain promises.